src/game_initialization/multiplayer.cpp

```diff
@@ -442,7 +401,9 @@ std::unique_ptr<wesnothd_connection> mp_manager::open_connection(std::string hos
 
 				const std::string ec = (*error)["error_code"];
 
-				if(ec == MP_MUST_LOGIN) {
+				if(is_pw_request && !conn->using_tls()) {
+					error_message = _("The remote server requested a password while using an insecure connection.");
```

Comment:
`is_pw_request` is also false if the password is empty even if there was a password request.

Reply:
Fixed.

Reply:
I would have actually fixed the variable instead. It has a misleading name that already tripped you up once.

Reply:
I decided to just delete the variable and update the if-statement's comments.